spynnaker/pyNN/models/neuron/synaptic_manager.py

```diff
@@ -461,7 +461,17 @@ def _get_ring_buffer_to_input_left_shifts(
                     spikes_per_second = self._spikes_per_second
                     if isinstance(app_edge.pre_vertex,
                                   SpikeSourcePoissonVertex):
-                        spikes_per_second = app_edge.pre_vertex.rate
+                        rate = app_edge.pre_vertex.rate
+                        # If non-zero rate then use it; otherwise keep default
+                        if hasattr(rate, "__getitem__"):
+                            sumrate = 0
+                            for i in range(len(rate)):
```

Comment:
probably easier to do sumrate = sum(rate)

Reply:
Definitely is - fixed.